src/main/java/org/openstreetmap/atlas/geography/atlas/raw/creation/RawAtlasGenerator.java

```diff
@@ -276,6 +281,21 @@ private void countOsmPbfEntities()
         logger.info("Counted PBF Entities in {}", countTime.elapsedSince());
     }
 
+    private long getLayerTagValueForPoint(final Atlas atlas, final long identifier)
+    {
+        final Optional<Long> layerValue = LayerTag.getTaggedOrImpliedValue(atlas.point(identifier),
```

Comment:
Thoughts on updating getTaggedOrImpliedValue to always return the implied value?  It looks like the only case where Optional.empty() can be returned is when the layer tag is populated but invalid.  If we fix that up, getTaggedOrImpliedValue doesn't need to return an optional anymore and will make this code (and other getLayerTag* calls) much simpler!

Reply:
I was thinking about that too when calling this, seemed a little clunky...my concern was in the case of a present, but invalid value. We can easily check for this and return... something like:

```
static Long getTaggedOrImpliedValue(final Taggable taggable, final Long impliedValue)
    {
        final Optional<String> tagValue = taggable.getTag(KEY);
        if (tagValue.isPresent())
        {
            final LongExtractor extractor = new LongExtractor();
            final Optional<Long> extractedValue = extractor.validateAndExtract(tagValue.get(),
                    LayerTag.class.getDeclaredAnnotation(Tag.class));
            if (extractedValue.isPresent())
            {
                return extractedValue.get();
            }
        }
        return impliedValue;
    }
```

However, getTaggedValue is relying on getTaggedOrImpliedValue. We can update getTaggedValue to return a Long as well, but then you would have to be doing a null check any time it's called. It seems like either a null check or Optional.empty() check is required here, without duplicating some code to remove getTaggedValue's dependency on getTaggedOrImpliedValue. I image getTaggedValue would look like (in which case you still have to do Optional check for this call):

```
static Optional<Long> getTaggedValue(final Taggable taggable)
    {
        final Optional<String> tagValue = taggable.getTag(KEY);
        if (tagValue.isPresent())
        {
            final LongExtractor extractor = new LongExtractor();
            return extractor.validateAndExtract(tagValue.get(),
                    LayerTag.class.getDeclaredAnnotation(Tag.class));
        }
        return Optional.empty();
    }
```

Thoughts?

Reply:
TLDR: might reduce overhead but cause some code duplication.

Reply:
Hmm great point!  Thinking about this more, I think we have the dependency in the wrong order.  I think getTaggedOrImpliedValue should depend on getTaggedValue and not the other way around.  Thoughts? (Haven't fully thought through all the side effects)

This would result in:
getTaggedValue:
```
        final Optional<String> tagValue = taggable.getTag(KEY);
        if (tagValue.isPresent())
        {
            final LongExtractor extractor = new LongExtractor();
            return extractor.validateAndExtract(tagValue.get(),
                    LayerTag.class.getDeclaredAnnotation(Tag.class));
        }
        return Optional.empty();
```

getTaggedOrImpliedValue:
```
static Long getTaggedOrImpliedValue(final Taggable taggable, final Long impliedValue)
    {
        final Optional<Long> taggedValue = getTaggedValue(taggable);
        return taggedValue.isPresent() ? taggedValue.get() : impliedValue;
    }
```

Reply:
I agree, this would work better!